jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+source "${JBOSS_HOME}/bin/launch/launch-common.sh"
+source "${JBOSS_HOME}/bin/launch/jboss-kie-common.sh"
+source "${JBOSS_HOME}/bin/launch/logging.sh"
+
+# make sure there is a workbench service within the namespace.
+serviceName=${WORKBENCH_SERVICE_NAME/-/_}
+serviceHost=$(find_env "${serviceName^^}_SERVICE_HOST")
+servicePort=$(find_env "${serviceName^^}_SERVICE_PORT_HTTP")
+
+update_config_map() {
+    # only execute the following lines if this container is running on OpenShift
+    if [ -e /var/run/secrets/kubernetes.io/serviceaccount/token ]; then
+        local namespace=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+        local token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+        local response=$(curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+            -H "Authorization: Bearer $token" \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/strategic-merge-patch+json' \
+            -XPATCH \
+            https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/api/v1/namespaces/${namespace}/configmaps/myapp-kieserver -d ${1} )
+    fi
+}
+
+if [ "${serviceHost}x" != "x" -a "${servicePort}x" != "x" ]; then
+    auth="$(echo -n "${KIE_ADMIN_USER}:${KIE_ADMIN_PWD}" | base64)"
+    if [ ! -z "${KIE_SERVER_ID}" ]; then
+        uri="deploymentconfigs?labelSelector=services.server.kie.org%2Fkie-server-id%3D${KIE_SERVER_ID}"
+        response=$(query_ocp_api "apis/apps.openshift.io" "${uri}")
+        replicas=$(echo ${response:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["items"][0]["spec"]["replicas"])')
+
+        bcResponse=$(query_ocp_api "apis/apps.openshift.io" "deploymentconfig/${WORKBENCH_SERVICE_NAME}")
+        bcReplicas=$(echo ${bcResponse:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["status"]["availableReplicas"])')
```

Comment:
I updated to readyReplicas, that should be enough.

Reply:
IMHO, both readyReplicas and availableReplicas are transient states, whereas spec.replicas defines a stable and final state. I'd rather rely on a stable state for making a decision on whether or not to make 'curl' call. For example, when a BC pod is spinning up (There will be a moment like this: spec.Replicas = 1, readyReplicas = 0, availableReplicas = 0), it is better to execute 'curl' and wait for a while. On the other hand, there may be a situation during BC scaling down where spec.Replicas = 0 but readyReplicas is still 1. In such case, no need to make 'curl' call as BC pod is shutting down anyway.

Reply:
well, if I understand correctly, status.readyReplicas is set only after probes reports that the service is up.
The spec.replicas is changed right way after you start the pod, I don't think it is a stable state.

>> here may be a situation during BC scaling down where spec.Replicas = 0 but readyReplicas is still 1

Not sure if it can happen.

Reply:
Or do you want to use the replicas and configure the curl command to wait a little bit for BC get up?

Reply:
Yes